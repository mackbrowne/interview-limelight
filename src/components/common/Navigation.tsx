// Framework
import React from 'react';
import { Link } from 'react-router-dom';

// Components
import { Navbar, Nav } from 'react-bootstrap';

export type NavItemType = {
  label: string,
  path: string
};

export interface NavProps {
  navigationItems: NavItemType[];
}

/**
 * A list of field rows with a header.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
export default function Navigation({ navigationItems = [] }: NavProps) {
  return (
    <Navbar color="faded" bg="light" expand="md">
      <Navbar.Brand href="/">Flickr Photo Viewer</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="ml-auto" navbar>
          {navigationItems.map(({ label, path }, i) => (
            <Nav.Link key={`nav-item-${i}`} as={Link} to={path}>
              {label}
            </Nav.Link>
          ))}
          <Nav.Item>
            <Nav.Link
              href="https://github.com/mackbrowne/react-bootstrap-webpack-starter/tree/interview-cygni"
              target="_blank"
            >
              Github
            </Nav.Link>
          </Nav.Item>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}
