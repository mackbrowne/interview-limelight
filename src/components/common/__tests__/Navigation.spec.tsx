import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import Navigation, { NavItemType } from '../Navigation';

describe('<Navigation /> Snapshot', () => {
  const render = (navigationItems: NavItemType[]) =>
    mount(
      <Router>
        <Navigation navigationItems={navigationItems} />
      </Router>
    );

  it('renders empty list', () => {
    const wrapper = render([]);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders links', () => {
    const wrapper = render([
      {
        label: 'Link1',
        path: '/link1'
      },
      {
        label: 'Link2',
        path: '/link2'
      },
      {
        label: 'Link3',
        path: '/link3'
      }
    ]);
    expect(wrapper).toMatchSnapshot();
  });
});
