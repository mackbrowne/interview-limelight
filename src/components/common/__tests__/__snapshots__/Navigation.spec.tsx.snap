// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snapshot renders empty list 1`] = `ReactWrapper {}`;

exports[`<Navigation /> Snapshot renders links 1`] = `ReactWrapper {}`;
